### ISSUE TITLE
Creation of S3 bucket is broken in v2.2.0

### ISSUE THREAD

**Issue by awconstable** [VerticeOne/terraform-aws-vertice-integration#13]
### Context
When using the vertice module with `cur_bucket_enabled = true` the terraform plan errors out due to the s3 bucket not being present. However the documentation recommends using the module in this mode which is supposed to create the S3 bucket as part of the plan.

### Terraform Configuration
```terraform
module "vertice_cco_integration_role" {
  source        = "git::https://github.com/VerticeOne/terraform-aws-vertice-integration.git?ref=v2.2.0"

  account_type = "billing"
  cur_bucket_enabled = true
  cur_report_enabled = true

  billing_policy_addons = {
    # allow managing EC2 Reserved Instances in billing policy
    ec2_ri = false
  }

  cur_bucket_name = "vertice-cur-reports-athena-${data.aws_caller_identity.current.account_id}"

  cur_report_name      = "athena"
  cur_report_s3_prefix = "cur"

  providers = {
    aws = aws

    aws.us-east-1 = aws.us-east-1
  }
}
```

### Result

Completing a terraform plan/apply results in the S3 bucket not being found

```shell
...
Plan: 12 to add, 1 to change, 0 to destroy.
╷
│ Error: Failed getting S3 bucket (vertice-cur-reports-athena-xxxxxxxxx): NotFound: Not Found
│       status code: 404, request id: xxxxx, host id: xxxxx
│ 
│   with module.vertice_cco_integration_role.module.vertice_cur_report[0].data.aws_s3_bucket.vertice_cur_bucket,
│   on .terraform/modules/vertice_cco_integration_role/modules/vertice-cur-report/main.tf line 1, in data "aws_s3_bucket" "vertice_cur_bucket":
│    1: data "aws_s3_bucket" "vertice_cur_bucket" {
│ 
╵
Releasing state lock. This may take a few moments...
```

### Expected Result

A clean plan and apply first time.

### Work around

Complete a targeted apply of the S3 bucket, followed by a regular apply.
```terraform
terraform apply -target "module.vertice_cco_integration_role.module.vertice_cur_bucket[0].module.vertice_cur_bucket.aws_s3_bucket.this[0]"
...
terraform apply
```

**Comment by gerricchaplin** [comment]
Hello Austen,

Thanks for raising this issue with us.
Sorry for the inconvenience this has caused.
We have created an internal ticket to remediate this.

I will let you know once a fix has been released.

Best regards,
Gerric Chaplin

**Comment by kristian-lesko** [comment]
Hello,

The fixed version has now been released as [v2.2.1](https://github.com/VerticeOne/terraform-aws-vertice-integration/releases/tag/v2.2.1).

Best regards,
Kristian Lesko
